### PR TITLE
Add hook weights to helm test resources where needed

### PR DESCRIPTION
**Pull request by imheresamir**
### What does this PR do?

- Raw manifest generator GitHub Action removed and to be superseded by #370
- The Helm test hooks are reintroduced to the chart test resources
- Hook weights are added to non-Pod resources such that these resources get created before the Pods do as a workaround for [this bug](https://github.com/helm/helm/issues/10006) in Helm which messes up the resource creation order
- Helm test scripts cleaned up a bit to remove usage of `helm test --logs` which has [a bug](https://github.com/helm/helm/issues/8342)

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

File: helm/conjur-config-cluster-prep/templates/tests/test-cluster-prep-configmap.yaml [hunk 1]
```diff
@@ -2,6 +2,9 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ .Release.Name }}-tests-configmap
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-weight": "-1"
```
Comment (diverdane) [inline helm/conjur-config-cluster-prep/templates/tests/test-cluster-prep-configmap.yaml:7]: Can you add a helpful reference to the Helm bug that explains why the `-1` hook weight is required into the PR description?

**Review by diverdane — COMMENTED** [review]
Looks good!!!
One comment, and a couple of questions:

- Should the template generation GH action script be deleted?
- Do we need a new GH action to check whether template generation needs to be done?

File: helm/conjur-config-cluster-prep/templates/tests/test-cluster-prep-configmap.yaml (hunk 1 above)
Comment (imheresamir) [reply helm/conjur-config-cluster-prep/templates/tests/test-cluster-prep-configmap.yaml:7]: Done

**Review by diverdane — APPROVED** [review]
LGTM!!!
Thanks for filing the followup issue, I take it we'll need to add a docker run of the generate command? I think there's already a test image to do that, although it's probably missing `yq`.

**Comment by imheresamir** [comment]
>  I take it we'll need to add a docker run of the generate command? I think there's already a test image to do that, although it's probably missing yq.
Yep! I'll add that to the new issue description :)

**Pull request merged**